diff --git a/modules/contrib/gvs/gvs.theme.inc b/modules/contrib/gvs/gvs.theme.inc
index d992e28..4786137 100644
--- a/modules/contrib/gvs/gvs.theme.inc
+++ b/modules/contrib/gvs/gvs.theme.inc
@@ -466,11 +466,8 @@ function template_preprocess_gvs_view_gvs(&$vars) {
       }
       $strdata .= "data.addRows(" . $no_of_rows . ");";
       for ($j=0;$j<$no_of_rows;$j++) {
-        $xfield=$view->field[$vars['options']['xaxis']]->real_field;
-        $xvalue=strip_tags($view->style_plugin->rendered_fields[$j][$xfield]);
-
-        $yfield=$view->field[$vars['options']['yaxis1_1']]->real_field;
-        $yvalue=strip_tags($view->style_plugin->rendered_fields[$j][$yfield]);
+        $xvalue=strip_tags($view->style_plugin->rendered_fields[$j][$vars['options']['xaxis']]);
+        $yvalue=strip_tags($view->style_plugin->rendered_fields[$j][$vars['options']['yaxis1_1']]);
 
         $strdata .= "data.setValue(" . $j . ", 0," . (is_null($xvalue)?'undefined':"'" .  str_replace("'","`",strip_tags($xvalue)) . "'") . ");";
 
@@ -520,16 +517,14 @@ function template_preprocess_gvs_view_gvs(&$vars) {
         $jjj=2;
         for ($jj=2;$jj<21;$jj++) {
           if ($vars['options']['yaxis' . $jj]!='') {
-            $yfield=$view->field[$vars['options']['yaxis' . $jj]]->real_field;
-            $yvalue=strip_tags($view->style_plugin->rendered_fields[$j][$yfield]);
+            $yvalue=strip_tags($view->style_plugin->rendered_fields[$j][$vars['options']['yaxis' . $jj]]);
             $strdata .= "data.setValue(" . $j . ", " . $jjj . ", " . (is_null($yvalue)?'undefined':$yvalue) . ");";
             $jjj++;
           }
         }
         for ($jj=2;$jj<21;$jj++) {
           if ($vars['options']['zaxis' . $jj]!='') {
-            $zfield=$view->field[$vars['options']['zaxis' . $jj]]->real_field;
-            $zvalue=strip_tags($view->style_plugin->rendered_fields[$j][$zfield]);
+            $zvalue=strip_tags($view->style_plugin->rendered_fields[$j][$vars['options']['zaxis' . $jj]]);
             $strdata .= "data.setValue(" . $j . ", " . $jjj . ", " . (is_null($zvalue)?'undefined':"'" .  str_replace("'","`",$zvalue) . "'") . ");";
             $jjj++;
           }
